### ISSUE TITLE
Period ligature bug

### ISSUE THREAD

**Issue by fnky** [larsenwork/monoid#112]
When typing more than 3 periods, it will show whitespace and a single ellipsis.

Text:
```
"   .           ..          ...       ........"
    ^           ^           ^         ^
full stop   double-dot   ellipsis    ellipsis followed
                                     by five full-stops
```

Xcode:
<img width="362" alt="dots" src="https://cloud.githubusercontent.com/assets/995050/9120005/238eec0a-3c7a-11e5-926e-04cca1e687e3.png">

**Comment by larsenwork** [comment]
what version are you running? should be fixed.

**Comment by fnky** [comment]
Updated to 0.60 and there's still issues with the periods. Take the following text

```
... .. .......... .. . ..d
```

<img width="448" alt="atom" src="https://cloud.githubusercontent.com/assets/995050/9133502/73ecb15a-3cfc-11e5-8a8c-c4d7dddd2d05.png">

As you can see, this even happens without ligatures enabled.

**Comment by larsenwork** [comment]
what happens? .. and ... are closer by design similar to || and soon !! since they denote a single "function" and end up too close the surrounding characters if they weren't squeezed a bit.

```
............... used to have bug that would make it look like 
            ... but that should have been fixed
```

**Comment by fnky** [comment]
Yes the example you gave is fixed in 0.60. I understand that they're closer, that is the ligature feature. But note the double period follow by d (`..d`). This shows whitespace between the dots (`.  .d`).

As you can see in my screenshot, this happens to both, no matter wether ligatures is on or off.

**Comment by larsenwork** [comment]
![screen shot 2015-08-08 at 22 07 52](https://cloud.githubusercontent.com/assets/3859425/9152038/f87b40a8-3e19-11e5-9451-9706f1cffb91.png)

I haven't been able to reproduce that - not sure what's going on. :/

**Comment by fnky** [comment]
Actually I tested it in atom, so it's a problem in Atom, which I know isn't supported. But somehow it changed the behavior from the above to this:

<img width="478" alt="atom" src="https://cloud.githubusercontent.com/assets/995050/9153629/234fbdbe-3e64-11e5-838d-d477455bcd44.png">

The cursor is simply a bug in Atom, not counting size of ligatures.

**Comment by larsenwork** [comment]
https://github.com/larsenwork/monoid/issues/112#issuecomment-129042037 I'm also using atom

**Comment by fnky** [comment]
This is the stylesheet I used

```scss
{
  -webkit-font-feature-settings: "liga" on, "calt" on;
}

atom-text-editor::shadow {
  .cursor-line {
    -webkit-font-feature-settings: "liga" off, "calt" off;
  }
}
```
It disables ligatures on current selected line.

**Comment by larsenwork** [comment]
I know - it's what I'm using too :) And Atom works just fine for me (Monoid - not Monoisome) 

Why is the last dot and the d orange? Maybe it's your syntax highlighting that's affecting it somehow...

**Comment by fnky** [comment]
Yep, it was using SCSS syntax highlighting. If there's syntax highlighting it will act as the above screenshot. However, if there isn't, the cursor will not be positioned according to the position in buffer.

![m](https://cloud.githubusercontent.com/assets/995050/9176907/426e0c70-3f8e-11e5-9eb4-7fabf87b61d4.png)

Note that this is tested with a generated font `Monoid HalfLoose`. I have yet to test other variations.
